connectors/sources/sharepoint_online.py

```diff
@@ -1365,6 +1379,11 @@ def _is_site_admin(user):
 
             access_control |= member_access_control
 
+        async for member in self.client.site_admins(site["webUrl"]):
```

Comment:
Reading the code above, it can be a bit confusing to the dev that we construct site_admins_access_control by iterating over 2 loops - both of them instinctively should return the same set of admins (I know they don't in reality).

Can you add a comment explaining that briefly for non-SPO experts 😄  ?